src/main/java/uk/gov/companieshouse/api/accounts/controller/StocksController.java

```diff
@@ -0,0 +1,186 @@
+                                 @PathVariable("companyAccountId") String companyAccountId,
+                                 HttpServletRequest request) {
+
+        SmallFull smallFull = (SmallFull) request.getAttribute(AttributeName.SMALLFULL.getValue());
+        if (smallFull.getLinks().get(SmallFullLinkType.STOCKS_NOTE.getLink()) == null) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+
+        if (bindingResult.hasErrors()) {
+            Errors errors = errorMapper.mapBindingResultErrorsToErrorModel(bindingResult);
+            return new ResponseEntity<>(errors, HttpStatus.BAD_REQUEST);
+        }
+
+        Transaction transaction = (Transaction) request.getAttribute(AttributeName.TRANSACTION.getValue());
+
+        try {
+            ResponseObject<Stocks> response = stocksService
+                    .update(stocks, transaction, companyAccountId, request);
+
+            return apiResponseMapper
+                    .map(response.getStatus(), response.getData(), response.getErrors());
+
+        } catch (DataException ex) {
+
+            final Map<String, Object> debugMap = createDebugMap(companyAccountId, transaction,
+                    "Failed to update stocks resource");
+            LOGGER.errorRequest(request, ex, debugMap);
+            return apiResponseMapper.map(ex);
+        }
+    }
+
+    @GetMapping
+    public ResponseEntity get(@PathVariable("companyAccountId") String companyAccountId,
+                              HttpServletRequest request) {
+
+        Transaction transaction = (Transaction) request
+                .getAttribute(AttributeName.TRANSACTION.getValue());
+
+        String stocksId = stocksService.generateID(companyAccountId);
+
+        ResponseEntity responseEntity;
+
+        try {
+            ResponseObject<Stocks> response = stocksService
+                    .findById(stocksId, request);
+
+            responseEntity = apiResponseMapper.mapGetResponse(response.getData(), request);
+
+        } catch (DataException de) {
+
+            final Map<String, Object> debugMap = createDebugMap(companyAccountId, transaction,
+                    "Failed to retrieve stocks resource");
+            LOGGER.errorRequest(request, de, debugMap);
+            responseEntity = apiResponseMapper.map(de);
+        }
+
+        return responseEntity;
+    }
+
+    @DeleteMapping
+    public ResponseEntity delete(@PathVariable("companyAccountId") String companyAccountsId,
+                                 HttpServletRequest request) {
+
+        SmallFull smallFull = (SmallFull) request.getAttribute(AttributeName.SMALLFULL.getValue());
+
+        Transaction transaction = (Transaction) request
+                .getAttribute(AttributeName.TRANSACTION.getValue());
+
+        String stocksId = stocksService.generateID(companyAccountsId);
+
+        try {
+            ResponseObject<Stocks> response = stocksService
+                    .deleteById(stocksId, request);
```

Comment:
You'll need to merge up from `develop`. The `ResourceService` interface now provides a `delete` method (formerly `deleteById`) which takes a `companyAccountsId`. This buys us the opportunity to call `SmallFullService#removeLink` within the delete method of the service to remove the resource link.